test/e2e/proxy_test.go

```diff
@@ -0,0 +1,193 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+	"net"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var hexDigit = []byte{'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f'}
+
+func hexString(b []byte) string {
+	s := make([]byte, len(b)*2)
+	for i, tn := range b {
+		s[i*2], s[i*2+1] = hexDigit[tn>>4], hexDigit[tn&0xf]
+	}
+	return strings.TrimLeft(string(s), "0")
+}
+
+func proxyEnabled(data *TestData, t *testing.T) (bool, error) {
+	key := "resubmit(,40),resubmit(,41)"
+	agentName, err := data.getAntreaPodOnNode(masterNodeName())
+	if err != nil {
+		return false, err
+	}
+	table31Output, _, err := data.runCommandFromPod(metav1.NamespaceSystem, agentName, "antrea-agent", []string{"ovs-ofctl", "dump-flows", defaultBridgeName, "table=31"})
+	return strings.Contains(table31Output, key), err
+}
+
+func TestProxyServiceSessionAffinity(t *testing.T) {
+	skipIfProviderIs(t, "kind", "Not work in kind test, need to be investigated.")
```

Comment:
```suggestion
	skipIfProviderIs(t, "kind", "Does not work in Kind, needs to be investigated.")
```

Let's open an issue to track this after this is merged (hopefully you can provide some additional details there). Do you think this could be because of an issue with the conntrack implementation of the OVS netdev datapath?

Reply:
Thanks, Antonin. I will create an issue for it. Since the hairpin test could pass, I don't think there is an issue about conntrack.

Reply:
Maybe something to do with learning then
Thanks for keeping track of this

Reply:
forgot to apply the suggestion of the message?

Reply:
Sorry, updated.